compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5696,13 +5639,17 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
             if (queryExpr.isStream) {
                 return new BStreamType(TypeTags.STREAM, selectType, completionType, null);
             } else if (queryExpr.isTable) {
-                actualType = getQueryTableType(queryExpr, selectType);
+                actualType = getQueryTableType(queryExpr, selectType, resolvedTypes.get(0), env);
```

Comment:
Why do we have resolvedTypes as a list and consider the 0th index as the correct resolved type?

Reply:
> Why do we have resolvedTypes as a list 

It was implemented by Grainier. I think, he used a list to keep all the resolved types compared to contextually expected type (unions etc.) and log ambiguous error when there were multiple resolved/selectTypes types. There is a known issue for this logic https://github.com/ballerina-platform/ballerina-lang/issues/36476.

> consider the 0th index as the correct resolved type?

Program will reach to this line only when the size of the resolvedTypes list is 1 (because both resolvedTypes and selectTypes lists have same length and size is checked in L5636). We have to pass the resolvedType to identify whether the contexually expected type is readonly type.